pkg/virtctl/adm/logverbosity/logverbosity.go

```diff
@@ -0,0 +1,447 @@
+	usage += "  {{ProgramName}} adm log-verbosity --virt-handler=7 --virt-launcher=3\n"
+
+	usage += "  # reset all components to default besides virt-handler which is 7\n"
+	usage += "  {{ProgramName}} adm log-verbosity --reset --virt-handler=7\n"
+	usage += "  # set all components to 3 besides virt-handler which is 7\n"
+	usage += "  {{ProgramName}} adm log-verbosity --all=3 --virt-handler=7\n"
+
+	return usage
+}
+
+// virtComponent to component name
+func getComponentNameByVirtComponent(component virtComponent) string {
+	var virtComponentToComponentName = map[virtComponent]string{
+		virtAPI:        "virt-api",
+		virtController: "virt-controller",
+		virtHandler:    "virt-handler",
+		virtLauncher:   "virt-launcher",
+		virtOperator:   "virt-operator",
+		all:            "all",
+	}
+	return virtComponentToComponentName[component]
+}
+
+// virtComponent to JSON name
+func getJSONNameByVirtComponent(component virtComponent) string {
+	var virtComponentToJSONName = map[virtComponent]string{
+		virtAPI:        "virtAPI",
+		virtController: "virtController",
+		virtHandler:    "virtHandler",
+		virtLauncher:   "virtLauncher",
+		virtOperator:   "virtOperator",
+		all:            "all",
+	}
+	return virtComponentToJSONName[component]
+}
+
+// component name to JSON name
+func getJSONNameByComponentName(componentName string) string {
+	var componentNameToJSONName = map[string]string{
+		"virt-api":        "virtAPI",
+		"virt-controller": "virtController",
+		"virt-handler":    "virtHandler",
+		"virt-launcher":   "virtLauncher",
+		"virt-operator":   "virtOperator",
+		"all":             "all",
+	}
+	return componentNameToJSONName[componentName]
+}
+
+func detectInstallNamespaceAndName(virtClient kubecli.KubevirtClient) (namespace, name string, err error) {
+	// listing KubeVirt CRs in all namespaces and see where it is installed
+	kvs, err := virtClient.KubeVirt("").List(&k8smetav1.ListOptions{})
+	if err != nil {
+		return "", "", fmt.Errorf("could not list KubeVirt CRs in all namespaces: %v", err)
+	}
+	if len(kvs.Items) == 0 {
+		return "", "", errors.New("could not detect a KubeVirt installation")
+	}
+	if len(kvs.Items) > 1 {
+		return "", "", errors.New("invalid kubevirt installation, more than one KubeVirt resource found")
+	}
+	namespace = kvs.Items[0].Namespace
+	name = kvs.Items[0].Name
+	return
+}
+
+func hasVerbosityInKV(kv *v1.KubeVirt) (map[string]uint, error) {
+	verbosityMap := map[string]uint{} // key: component name, value: verbosity
+	// check the logVerbosity field in the KubeVirt CR
+	lvJSON, err := json.Marshal(kv.Spec.Configuration.DeveloperConfiguration.LogVerbosity)
+	if err != nil {
+		return nil, err
+	}
+	// if there is a logVerbosity field, input the JSON data to the verbosityMap
+	if err := json.Unmarshal(lvJSON, &verbosityMap); err != nil {
+		return nil, err
+	}
+	return verbosityMap, nil
+}
+
+func createOutputMessage(verbosityVal map[string]uint, options map[string]uint) []string {
+	var messages []string
```

Comment:
I'd actually use `lines` instead of `messages`

Reply:
Agreed.